lsp-diagnostics.el

```diff
@@ -0,0 +1,315 @@
+  "Report new diagnostics to flymake."
+  (funcall lsp-diagnostics--flymake-report-fn
+           (-some->> (lsp-diagnostics t)
+             (gethash (lsp--fix-path-casing buffer-file-name))
+             (--map (-let* (((&Diagnostic :message :severity?
+                                          :range (range &as &Range
+                                                        :start (&Position :line start-line :character)
+                                                        :end (&Position :line end-line))) it)
+                            ((start . end) (lsp--range-to-region range)))
+                      (when (= start end)
+                        (if-let ((region (flymake-diag-region (current-buffer)
+                                                              (1+ start-line)
+                                                              character)))
+                            (setq start (car region)
+                                  end (cdr region))
+                          (lsp-save-restriction-and-excursion
+                            (goto-char (point-min))
+                            (setq start (point-at-bol (1+ start-line))
+                                  end (point-at-eol (1+ end-line))))))
+                      (flymake-make-diagnostic (current-buffer)
+                                               start
+                                               end
+                                               (cl-case severity?
+                                                 (1 :error)
+                                                 (2 :warning)
+                                                 (t :note))
+                                               message))))
+           ;; This :region keyword forces flymake to delete old diagnostics in
+           ;; case the buffer hasn't changed since the last call to the report
+           ;; function. See https://github.com/joaotavora/eglot/issues/159
+           :region (cons (point-min) (point-max))))
+
+
+
+;;;###autoload
+(defun lsp-diagnostics--enable ()
+  "Enable LSP checker support."
+  (unless (eq lsp-diagnostics-provider :none)
+    (lsp-diagnostics-mode 1)))
+
+(defun lsp-diagnostics--disable ()
+  "Disable LSP checker support."
+  (lsp-diagnostics-mode -1))
+
+;;;###autoload
+(define-minor-mode lsp-diagnostics-mode
+  "Toggle LSP diagnostics integration."
+  :group 'lsp-mode
+  :global nil
+  :lighter ""
+  (cond
+   (lsp-diagnostics-mode
+    (cond
+     ((or
+       (and (eq lsp-diagnostics-provider :auto)
+            (functionp 'flycheck-mode))
+       (and (eq lsp-diagnostics-provider :flycheck)
+            (or (functionp 'flycheck-mode)
+                (user-error "The lsp-diagnostics-provider is set to :flycheck but flycheck is not installed?")))
+       ;; legacy
+       (null lsp-diagnostics-provider))
+      (lsp-diagnostics--flycheck-enable))
+     ((and (not (version< emacs-version "26.1"))
```

Comment:
Can this version check be removed? I think we all upgrade to `26.x` now for `lsp-mode`

Reply:
Done!